lib/plugins/physics.js

```diff
@@ -330,30 +345,59 @@ function inject (bot, { physicsEnabled, maxCatchupTicks }) {
     await bot.look(yaw, pitch, force)
   }
 
+  // 1.21.3+
+  bot._client.on('player_rotation', (packet) => {
+    bot.entity.yaw = conv.fromNotchianYaw(packet.yaw)
+    bot.entity.pitch = conv.fromNotchianPitch(packet.pitch)
+  })
+
   // player position and look (clientbound)
   bot._client.on('position', (packet) => {
     // Is this necessary? Feels like it might wrongly overwrite hitbox size sometimes
     // e.g. when crouching/crawling/swimming. Can someone confirm?
     bot.entity.height = 1.8
 
-    // Velocity is only set to 0 if the flag is not set, otherwise keep current velocity
+    // Velocity is reset if the x, y, z flags are not set
     const vel = bot.entity.velocity
-    vel.set(
-      packet.flags & 1 ? vel.x : 0,
-      packet.flags & 2 ? vel.y : 0,
-      packet.flags & 4 ? vel.z : 0
-    )
-
-    // If flag is set, then the corresponding value is relative, else it is absolute
+    // If the x, y, z flags are not set, the position is absolute
     const pos = bot.entity.position
-    pos.set(
-      packet.flags & 1 ? (pos.x + packet.x) : packet.x,
-      packet.flags & 2 ? (pos.y + packet.y) : packet.y,
-      packet.flags & 4 ? (pos.z + packet.z) : packet.z
-    )
-
-    const newYaw = (packet.flags & 8 ? conv.toNotchianYaw(bot.entity.yaw) : 0) + packet.yaw
-    const newPitch = (packet.flags & 16 ? conv.toNotchianPitch(bot.entity.pitch) : 0) + packet.pitch
+
+    // TODO: this current mineflayer logic maybe incorrect. New (maybe even older) versions of minecraft have flag values for
+    // dx/dy/dz but mineflayer is assuming that 0b111 applies for both velocity and position.
+
+    let newYaw, newPitch
+
+    if (bot.registry.version['>=']('1.21.3')) {
```

Comment:
Should be a feature

Reply:
The below features can be removed after https://github.com/PrismarineJS/minecraft-data/pull/948 so if there is a feature added here it would just be temporary